src/win/process.c

```diff
@@ -1250,7 +1253,82 @@ static int uv__kill(HANDLE process_handle, int signum) {
     return UV_EINVAL;
   }
 
+  /*
+   * Create a dump file for the targeted process, if the registry key
+   * `HKLM:Software\Microsoft\Windows\Windows Error Reporting\LocalDumps\DumpFolder`
+   * points to a valid folder.  The default value if the registry key does not
+   * exist is `%LOCALAPPDATA%\CrashDumps`, see [0] for more detail.
+   * [0]: https://learn.microsoft.com/en-us/windows/win32/wer/collecting-user-mode-dumps
+   */
+  if (signum == SIGQUIT) {
+    /* Get target process name. */
+    WCHAR basename[MAX_PATH];
+    GetModuleBaseNameW(process_handle, NULL, &basename[0], sizeof(basename));
+
+    /* Get PID of target process. */
+    DWORD pid = GetProcessId(process_handle);
+
+    /* Get LocalDumps directory path. */
+    HKEY registry_key;
+    WCHAR dump_folder[MAX_PATH];
+    DWORD ret = RegOpenKeyExW(HKEY_LOCAL_MACHINE,
+                  L"SOFTWARE\\Microsoft\\Windows\\Windows Error Reporting\\LocalDumps",
+                  0,
+                  KEY_QUERY_VALUE,
+                  &registry_key);
+    if (ret == ERROR_SUCCESS) {
+      DWORD dump_folder_len = sizeof(dump_folder);
+      DWORD key_type = 0;
+      ret = RegGetValueW(registry_key,
+                          NULL,
+                          L"DumpFolder",
+                          RRF_RT_ANY,
+                          &key_type,
+                          (PVOID) dump_folder,
+                          &dump_folder_len);
+      if (ret != ERROR_SUCCESS) {
+        /* Default value for `dump_folder` is `%LOCALAPPDATA%\CrashDumps`. */
+        WCHAR * localappdata;
+        SHGetKnownFolderPath(&FOLDERID_LocalAppData, 0, NULL, &localappdata);
+        _snwprintf_s(dump_folder, sizeof(dump_folder), _TRUNCATE, L"%ls\\CrashDumps", localappdata);
+        CoTaskMemFree(localappdata);
+      }
+      RegCloseKey(registry_key);
+    }
+
+    /* Construct dump filename. */
+    WCHAR dump_name[MAX_PATH];
+    _snwprintf_s(dump_name, sizeof(dump_name), _TRUNCATE, L"%ls\\%ls.%d.dmp", dump_folder, basename, pid);
+
+    HANDLE hDumpFile = CreateFileW(dump_name, GENERIC_WRITE, 0, NULL, CREATE_NEW, FILE_ATTRIBUTE_NORMAL, NULL);
```

Comment:
@vtjnash I was doing some more strenuous testing of this, and I discovered that WER actually is a little more aggressive than us; it will try to create the `dump_folder` directory if it does not exist.

Also, the gate for WER to create a local dump at all is the presence of the `LocalDumps` registry key.

My proposed behavior change here is to:

1) Check that `HKLM:SOFTWARE\Microsoft\Windows\Windows Error Reporting\LocalDumps` exists at all, and if it does not, break.
2) Try to create the crash dumps folder if it doesn't already exist.

Reply:
I think I am okay with deviating slightly there, if useful. But (1) is definitely needed, since it appears you are accessing dump_folder undefined here

Reply:
Alright, I figured out how to properly emulate WER here, and it's quite simple.  I think we've got full compatibility now, as far as this goes.  We're not emulating the process-name-specific overrides, but I'm okay with leaving that for future work.  :)